generators/generator-base.js

```diff
@@ -1196,7 +1196,15 @@ Generator.prototype.getConstraintName = function (entityName, relationshipName,
         var halfLimit = Math.floor(limit/2),
             entityTable = noSnakeCase ? entityName : this.getTableName(entityName.substring(0, halfLimit)),
             relationTable = noSnakeCase ? relationshipName: this.getTableName(relationshipName.substring(0, halfLimit - 1));
-        return `${entityTable}_${relationTable}_id`;
+
+        constraintName = `${entityTable}_${relationTable}_id`;
```

Comment:
the correct fix here should be changing `entityTable = noSnakeCase ? entityName : this.getTableName(entityName.substring(0, halfLimit))` to `entityTable = noSnakeCase ? entityName.substring(0, halfLimit) : this.getTableName(entityName.substring(0, halfLimit))` and changing `relationTable = noSnakeCase ? relationshipName: this.getTableName(relationshipName.substring(0, halfLimit - 1));` to `relationTable = noSnakeCase ? relationshipName.substring(0, halfLimit - 1) : this.getTableName(relationshipName.substring(0, halfLimit - 1));`